pex/resolve/resolver_options.py

```diff
@@ -135,8 +135,25 @@ def register(
             "See: https://pip.pypa.io/en/stable/topics/authentication/#keyring-support"
         ),
     )
+
     register_use_pip_config(parser)
 
+    parser.add_argument(
+        "--keyring-provider",
+        metavar="PROVIDER",
+        dest="keyring_provider",
+        type=str,
+        default=None,
+        help=(
+            "Configure Pip to use the given keyring provider to obtain authentication for package indexes. "
```

Comment:
@jsirois: I expanded the help text here.

Should the docs site have a place discussing how to use keyring?

Reply:
Probably, buts that's true of alot of things. I'm fine with you just landing this.